internal/pkg/instrumentation/probe/probe.go

```diff
@@ -36,17 +33,36 @@ type Probe interface {
 	// the information about the package the Probe instruments.
 	Manifest() Manifest
 
-	// Load loads all the eBPF programs and maps required by the Probe.
-	// It also attaches the eBPF programs to the target process.
-	// TODO: currently passing Sampler as an initial configuration - this will be
-	// updated to a more generic configuration in the future.
-	Load(*link.Executable, *process.Info, *sampling.Config) error
+	// Init initializes the Probe, setting up the reader, closers, and sampling config.
+	Init(*sampling.Config) error
 
 	// Run runs the events processing loop.
 	Run(func(ptrace.ScopeSpans))
 
 	// Close stops the Probe.
 	Close() error
+
+	// GetLogger returns the *slog.Logger associated with the Probe.
+	GetLogger() *slog.Logger
+
+	// Spec returns the *ebpf.CollectionSpec for the Probe.
+	Spec() (*ebpf.CollectionSpec, error)
+
+	// SetCollection sets the *ebpf.Collection for the Probe.
+	SetCollection(*ebpf.Collection)
+
+	// GetCollection returns the *ebpf.Collection for the Probe.
+	GetCollection() *ebpf.Collection
```

Comment:
This has a smell to it. Information and processing should flow in one direction. Why would we need to get and set this value. Can it be avoided?

Reply:
after the sig call, I am moving the collection tracking to the manager, so these aren't necessary for the probe anymore